api/src/snapshots/resolvers/veleroClient.ts

```diff
@@ -446,28 +452,27 @@ export class VeleroClient {
     };
 
     switch (store.provider) {
-    case SnapshotProvider.S3AWS:
-      const {accessKeyID, accessKeySecret} = await readAWSCredentialsSecret(corev1, this.ns);
-
-      store.s3AWS = {
-        region: bsl.spec.config.region,
-        accessKeyID,
-      };
-      if (accessKeySecret) {
-        store.s3AWS.accessKeySecret = redacted;
-      }
-      break;
-
-      case SnapshotProvider.S3Compatible:
-        const s3Creds = await readAWSCredentialsSecret(corev1, this.ns);
-
-        store.s3Compatible = {
-          region: bsl.spec.config.region,
-          endpoint: bsl.spec.config.endpoint,
-          accessKeyID: s3Creds.accessKeyID,
-        };
-        if (s3Creds.accessKeySecret) {
-          store.s3Compatible.accessKeySecret = redacted;
+      case SnapshotProvider.S3AWS:
```

Comment:
The velero BackupStorageLocation uses "aws" as provider for all s3-compatible endpoints. We distinguish between  S3 and other compatible stores in our SnapshotStore type so we can show show an input field for the endpoint for the s3-compatible stores.